src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java

```diff
@@ -552,6 +554,27 @@ private static AuthToken getAweAdminAuth(Map<String, String> config)
         return cachedAweAdminAuth;
     }
 	
+    public static CheckJobCanceledResult checkJobCanceled(
+            final CancelJobParams params,
+            final AuthToken authPart,
+            final Map<String, String> config) throws Exception {
+        final String ujsUrl = config.get(NarrativeJobServiceServer.CFG_PROP_JOBSTATUS_SRV_URL);
+        if (params == null) {
+            throw new NullPointerException("No parameters supplied to method");
+        }
+        final String jobId = params.getJobId();
+        if (jobId == null || jobId.trim().isEmpty()) {
+            throw new IllegalArgumentException("No job id supplied");
+        }
+        final UserAndJobStateClient ujsClient = getUjsClient(authPart, config);
+        final Tuple7<String, String, String, Long, String, Long, Long> jobStatus = 
+                ujsClient.getJobStatus(jobId);
+        return new CheckJobCanceledResult().withJobId(jobId).withUjsUrl(ujsUrl)
+                // null if job not started yet
+                .withFinished(jobStatus.getE6() == null ? 0 : jobStatus.getE6())
+                .withCanceled(APP_STATE_CANCELED.equals(jobStatus.getE2()) ? 1L : 0L);
```

Comment:
If the job is associated with a workspace then anyone who can write to that workspace can cancel the job.